docs/reference/mapping/types/runtime.asciidoc

```diff
@@ -0,0 +1,585 @@
+<<modules-scripting-using,define a Painless script>>, which has access to the
+entire context of a document, the original `_source`, and the mapped field plus
+its values (`doc_values`). At search time, the script runs and generates values
+for each scripted field.
+
+Runtime fields are similar to the <<script-fields,`script_fields`>> parameter
+of the `_search` request. You can retrieve the results of running a script, but
+also make the script results available for queries and aggregations.
+
+[[runtime-updating-scripts]]
+.Updating runtime scripts
+****
+
+Updating a script while a dependent query is running can return
+inconsistent results. Each shard might have access to different versions of the
+script, depending on when the mapping change takes effect.
+
+Existing queries or visualizations in {kib} that rely on runtime fields can
+fail if you change the `runtime_type`. For example, a bar chart visualization
+that uses a runtime field of `ip` will fail if the `runtime_type` is changed
+to `boolean`.
+
+****
+
+The script in the following request extracts the day of the week from the
+`@timestamp` field, which is defined as a `date` data type.
+
+[source,console]
+----
+PUT /my-index
+{
+  "mappings": {
+    "runtime": { <1>
+      "day_of_week": {
+        "type": "keyword", <2>
+        "script": { <3>
+          "source": "emit(doc['@timestamp'].value.dayOfWeekEnum.getDisplayName(TextStyle.FULL, Locale.ROOT))"
+        }
+      }
+    },
+    "properties": {
+      "timestamp": {"type": "date"}
+    }
+  }
+}
+----
+
+<1> Runtime fields are defined in the `runtime` section of the mapping
+definition.
+<2> Each runtime has its own field type, just like any other field.
+<3> The script defines the evaluation to calculate at search time.
+
+[[runtime-search-request]]
+==== Defining runtime fields in a search request
+You can specify a `runtime_mappings` section in a search request to create
+runtime fields that exist only as part of the query. You must specify a script
+as part of the `runtime_mappings` section, just as you would if adding a
+runtime field to the mappings.
+
+In the following request, the values for the `day_of_week` field are calculated
+dynamically, and only within the context of this search request.
+
+[source,console]
+----
+GET my-index/_search
+{
+  "runtime_mappings": {
+    "day_of_week": {
+      "type": "keyword",
+      "script": {
+        "source": "emit(doc['@timestamp'].value.dayOfWeekEnum.getDisplayName(TextStyle.FULL, Locale.ROOT))"
+      }
+    }
+  },
+  "aggs": {
+    "day_of_week": {
+      "terms": {
+        "field": "day_of_week"
+      }
+    }
+  }
+}
+----
+// TEST[continued]
+
+[[runtime-overriding-fields]]
+===== Overriding fields in the search request
+You can use the `runtime_mappings` section of the `_search` request to override
+field values inside objects by naming the runtime fields with dot notation. For
+example, index the following documents into `my-index`.
```

Comment:
I think I see what you are after here, but I find this section slightly confusing, because the dot notation can also be used to specify fields that are not at the root level, regardless of whether they are also defined under properties.

Reply:
I changed this section to **Shadowing fields in the search request**, and added the following:

> If you create a runtime field with the same name as a field that
> already exists in the mapping, the runtime field shadows the mapped field. At
> search time, {es} calculates the value of the runtime field and returns it as
> part of the query. Because the runtime field shadows the mapped field, you can
> modify the value returned in search without modifying the mapped field.
> 
> Objects are not supported in the `"runtime"` section under the
> mapping definition. However, you can use the `runtime_mappings` section of the
> `_search` request to shadow field values inside objects.